File: FeedBuilder/ArgumentsParser.vb
Imports System.IO
Imports System.Text.RegularExpressions

Public Class ArgumentsParser
    Public Property HasArgs As Boolean
    Public Property FileName As String
    Public Property ShowGui As Boolean
    Public Property Build As Boolean
    Public Property OpenOutputsFolder As Boolean

    Public Sub New(args As String())
        For Each arg As String In args
            If arg = Application.ExecutablePath Then Continue For

            arg = CleanArg(arg)
            If arg = "build" Then
                Me.Build = True
                Me.HasArgs = True
            ElseIf arg = "showgui" Then
                Me.ShowGui = True
                Me.HasArgs = True
            ElseIf arg = "openoutputs" Then
                Me.OpenOutputsFolder = True
                Me.HasArgs = True
            ElseIf File.Exists(arg) Then
                Me.FileName = arg
                Me.HasArgs = True
            Else
                Console.WriteLine("Unrecognized arg '{0}'", arg)
            End If

        Next
    End Sub

    Private Function CleanArg(arg As String) As String
        Const pattern1 As String = "^(.*)([=,:](true|0))"
        arg = arg.ToLower()
        If arg.StartsWith("-") OrElse arg.StartsWith("/") Then
            arg = arg.Substring(1)
        End If
        Dim r As New Regex(pattern1)
        arg = r.Replace(arg, "{$1}")
        Return arg
    End Function
End Class

File: FeedBuilder/ArgumentsParser.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Text.RegularExpressions;
using System.Windows.Forms;

namespace FeedBuilder
{
	public class ArgumentsParser
	{
		public bool HasArgs { get; set; }
		public string FileName { get; set; }
		public bool ShowGui { get; set; }
		public bool Build { get; set; }
		public bool OpenOutputsFolder { get; set; }

		public ArgumentsParser(IEnumerable<string> args)
		{
			foreach (string thisArg in args) {
				if (thisArg.ToLower() == Application.ExecutablePath.ToLower() || thisArg.ToLower().Contains(".vshost.exe")) continue;

				string arg = CleanArg(thisArg);
				if (arg == "build") {
					Build = true;
					HasArgs = true;
				} else if (arg == "showgui") {
					ShowGui = true;
					HasArgs = true;
				} else if (arg == "openoutputs") {
					OpenOutputsFolder = true;
					HasArgs = true;
				} else if (IsValidFileName(thisArg)) {
					// keep the same character casing as we were originally provided
					FileName = thisArg;
					HasArgs = true;
				} else Console.WriteLine("Unrecognized arg '{0}'", arg);
			}
		}

		// this merely checks whether the parent folder exists and if it does, 
		// we say the filename is valid
		private bool IsValidFileName(string filename)
		{
			if (File.Exists(filename)) return true;
			try {
				// the URI test... filter out things that aren't even trying to look like filenames
// ReSharper disable UnusedVariable
				Uri u = new Uri(filename);
// ReSharper restore UnusedVariable
				// see if the arg's parent folder exists
				var d = Directory.GetParent(filename);
				if (d.Exists) return true;
			} catch {}
			return false;
		}

		private string CleanArg(string arg)
		{
			const string pattern1 = "^(.*)([=,:](true|0))";
			arg = arg.ToLower();
			if (arg.StartsWith("-") || arg.StartsWith("/")) arg = arg.Substring(1);
			Regex r = new Regex(pattern1);
			arg = r.Replace(arg, "{$1}");
			return arg;
		}
	}
}